playwright-e2e/tests/reject-a-requested-visit.spec.ts

```diff
@@ -0,0 +1,149 @@
+            null,
+            false,
+            'Automation Tests',
+            sessionEndTime
+        )
+
+        console.log('Template ID:', templateId)
+
+        expect(templateStatus).toBe(201)
+        expect(templateId).toBeTruthy()
+        GlobalData.set('templateId', templateId)
+
+        // 2. Create application
+        const application: IApplication = {
+            prisonCode: prisonerDetails.prisonCode,
+            prisonerId: prisonerDetails.prisonerId,
+            sessionDate: format(sessionSlotTime, 'yyyy-MM-dd'), // match template date
+            sessionStart: '16:25:00',                            // match template start
+            sessionEnd: '18:25:00',                              // match template end
+            userType: 'STAFF',
+            contactName: 'IC',
+            visitors: [prisonerDetails.visitorId],
+            visitRestriction: 'OPEN',
+        }
+
+        const createApplicationResponse = await createApplication({ request }, application)
+        expect(createApplicationResponse.status).toBe(200)
+
+        const applicationRef = createApplicationResponse.applicationRef
+        expect(applicationRef).toBeTruthy()
+        GlobalData.set('applicationReference', applicationRef)
+
+        await testInfo.attach('Application created', {
+            body: `Application reference: ${applicationRef}`,
+            contentType: 'text/plain',
+        })
+
+        // 3. Book visit
+        const createVisitResponse = await createVisit({ request }, applicationRef)
+        expect(createVisitResponse.status).toBe(200)
+        expect(createVisitResponse.visitRef).toBeTruthy()
+
+        GlobalData.set('visitReference', createVisitResponse.visitRef)
+
+        await testInfo.attach('Visit created', {
+            body: `Visit reference: ${createVisitResponse.visitRef}`,
+            contentType: 'text/plain',
+        })
+
+        // ---------------- UI FLOW ----------------
+        await loginPage.navigateTo('/')
+        await loginPage.checkOnPage('HMPPS Digital Services - Sign in')
+        await loginPage.signInWith(UserType.USER_ONE)
+
+        await homePage.displayBookOrChangeaVisit()
+        await homePage.checkOnPage('Manage prison visits - DPS')
+        await homePage.clickOnRequestedVisits()
+        expect(await requestedVisitsPage.checkOnPage("Requested visits - Manage prison visits - DPS"))
+        await requestedVisitsPage.clickViewLinkForPrisoner(prisonerDetails.prisonerId)
+        await visitDetailsPage.rejectVisit()
+        expect(await requestedVisitsPage.verifyAlertText('You rejected the request to visit'))
+
+        const visitReference = GlobalData.get('visitReference')
+        expect(visitReference).toBeTruthy()
+        await testInfo.attach('Visit searched in UI', {
+            body: `Verified visit with reference: ${visitReference}`,
+            contentType: 'text/plain',
+        })
+    })
+
+    test.afterAll('Teardown', async ({ request }, testInfo) => {
```

Comment:
I think it will be handy adding a check before and after on the prisoner profile page that the visit status is showing REQUESTED before and CANCELLED after rejection?